clang/lib/Driver/Driver.cpp

```diff
@@ -6677,6 +6677,9 @@ const ToolChain &Driver::getToolChain(const ArgList &Args,
       TC = std::make_unique<toolchains::AMDGPUToolChain>(*this, Target, Args);
       break;
     case llvm::Triple::UEFI:
+      if (Target.getArch() != llvm::Triple::x86_64)
+        llvm::report_fatal_error("Currently the only architecture supported by "
```

Comment:
Pleasue use the diagnostics to output an error, instead of `report_fatal_error`. I'd guess that `computeTargetTriple` may be a better place to do such a check.  AIX is emitting an error there, that seems fairly close to your use case.